In [1]:
import pandas as pd

data = pd.read_csv('datasets/boston_housing.csv')
data
from sklearn.model_selection import train_test_split

x = data.drop('MEDV', axis=1)
y = data['MEDV']

x_train, x_test, y_train, y_test = train_test_split(x,y, test_size=0.25, random_state=0)
from sklearn.ensemble import RandomForestRegressor

model = RandomForestRegressor(n_estimators=200 ,max_depth=14, criterion='squared_error', random_state=69)
model.fit(x_train, y_train)

y_pred = model.predict(x_test)

from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

mse = mean_squared_error(y_pred, y_test)
mae = mean_absolute_error(y_pred, y_test)
r2 = r2_score(y_pred,y_test)

print("MSE: ", mse)
print("MAE: ", mae)
print("R2:", r2)

MSE:  15.649821542808082
MAE:  2.515428921809782
R2: 0.7367612717883111


---
## RandomForestRegressor()

**config:** n_estimators=200 ,max_depth=14, criterion='squared_error', random_state=69

**MSE:**  15.649821542808082  
**MAE:**  2.515428921809782  
**R2:** 0.7367612717883111

---

## LinearRegression()

**MSE:** 18.60219336052116  
**MAS:** 3.081985413223105  
**R^2:** 0.715995216261967

---